tracer/src/Datadog.Trace/Ci/CIVisibility.cs

```diff
@@ -157,41 +157,47 @@ private static async Task ShutdownAsync()
 
         private static bool InternalEnabled()
         {
+            var processName = ProcessHelpers.GetCurrentProcessName() ?? string.Empty;
+
+            // By configuration
             if (_settings.Enabled)
             {
+                // When is enabled by configuration we only enable it to the testhost child process if the process name is dotnet.
+                if (processName.Equals("dotnet", StringComparison.OrdinalIgnoreCase) && Environment.CommandLine.IndexOf("testhost.dll", StringComparison.OrdinalIgnoreCase) == -1)
+                {
+                    Log.Information("CI Visibility disabled because the commandline doesn't contains testhost.dll when the process name is dotnet: {cmdline}", Environment.CommandLine);
```

Comment:
```suggestion
                    Log.Information("CI Visibility disabled because the process name is 'dotnet' but the commandline doesn't contain 'testhost.dll': {cmdline}", Environment.CommandLine);
```